src/linter_test.cc

```diff
@@ -62,5 +62,25 @@ TEST(NolintTest, MultipleDisabling) {
   EXPECT_TRUE(errors[3].GetPosition() == std::make_pair(8, 2));
 }
 
+TEST(LinterTest, StatementValidityCheck) {
+  LinterOptions option;
+  EXPECT_TRUE(CheckParserSucceeds("SELECT 5+2", &option).ok());
```

Comment:
Yes, it accumulates, but that will never be used. I added a comment for this behavior.

For another thought: CheckParserSucceeds just adds an optimization flag and AST to options. So even when if it wasn't invoked, linter will be normal but slower since multiple parser calls will be made. I currently made "RunChecks" so that CheckParserSucceeds will run before all checks.

Reply:
What happens if the user disables parser-succeeds check? Can you extract file parsing in a completely separate function called before all checks? It could store the parser output or the error if there was a parsing error. Then if CheckParserSucceeds is enabled, it would just check if you have stored parser output and no parsing errors. Also, this will make the order of checks irrelevant.

Reply:
"CheckParserSucceeds" is completely seperate function that doesn't count as a "check" since does not follow the signature and of course it can't be in checks_list. This function will store parser output, or return an error if there was a parsing error. It can also be enabled/disabled. If it is enabled and parser succeeds, Parser-Dependant checks will work fine, otherwise Parser-Dependant Checks will also be skipped.

I think clang-tidy/cpplint only prints actual fixable lint errors. So, there shouldn't be skipped message for each check when parser is disabled or failed. It should just ignore, which is the current implementation.

Reply:
ah, ok, sounds good!